tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/IRChecker.scala

```diff
@@ -1060,9 +1060,16 @@ class IRChecker(unit: LinkingUnit, logger: Logger) {
 }
 
 object IRChecker {
-  private final class ErrorContext private (pos: Position, name: String) {
-    override def toString(): String =
+  private final class ErrorContext private (val treeOrLinkedClass: Any)
+      extends AnyVal {
+
+    override def toString(): String = {
+      val (pos, name) = treeOrLinkedClass match {
```

Comment:
Is this tuple supposed to get stack allocated on the JVM?
If not, it might be better to lift the `s"..."` to an inner function and calling it in place of the tuple creation.